File: src/2937.make-three-strings-equal.3207/solution.ts
function findMinimumOperations (s1: string, s2: string, s3: string): number {
    if (s1[0] !== s2[0] || s2[0] !== s3[0]) {
        return -1;
    }
    const MAX_INDEX = Math.min(s1.length, s2.length, s3.length);
    let index = 0;
    while (index < MAX_INDEX && s1[index] === s2[index] && s2[index] === s3[index]) {
        index++;
    }

    return s1.length + s2.length + s3.length - 3 * index;
}

File: src/2937.make-three-strings-equal.3207/solution.java
class Solution {
    public int findMinimumOperations(String s1, String s2, String s3) {
        char[] chars1 = s1.toCharArray();
        char[] chars2 = s2.toCharArray();
        char[] chars3 = s3.toCharArray();
        if(chars1[0] != chars2[0] || chars2[0] != chars3[0]){
            return -1;
        }
        int L = Math.min(Math.min(chars1.length,chars2.length),chars3.length);
        int index = 1;
        while(index<L){
            if(chars1[index] != chars2[index] || chars2[index] != chars3[index]){
                break; 
            }
            index++;
        }
        
        return chars1.length+chars2.length+chars3.length-index*3;
    }
}